synapse/handlers/sync.py

```diff
@@ -982,6 +982,10 @@ def _get_rooms_changed(self, sync_result_builder, ignored_users):
             # we do send down the room, and with full state, where necessary
             old_state_ids = None
             if room_id in joined_room_ids or has_join:
+                # Always include if the user (re)joined the room
```

Comment:
can you expand the comment to explain why this is the same thing as the condition? why does `non_joins` mean this?